lib/mix/lib/mix/tasks/escript.build.ex

```diff
@@ -136,14 +136,6 @@ defmodule Mix.Tasks.Escript.Build do
     filename = escript_opts[:path] || script_name
     main = escript_opts[:main_module]
 
-    unless script_name do
-      error_message =
-        "Could not generate escript, no name given, " <>
-          "set :name escript option or :app in the project settings"
-
-      Mix.raise(error_message)
-    end
```

Comment:
Dialyzer suggests this is dead code and I agree, `script_name` can not be nil and there is no test for this error. But it could be an empty string (maybe?), so possibly this should be changed to check for that?

Reply:
We can remove this. :) Good change. :+1: